### PR TITLE
updated blacklist/whitelist for pasted html elements

### PR DESCRIPTION
**Pull request by DZGoldman**
Went over a comprehensive list of html tags with @dginev and updated accordingly. 

Blacklisted: element is removed entirely
Allowed: element is kept as is
Neither: Element is unwrapped (tag is removed and contents get cleaned)

### CODE REVIEW

File: source/Clean.js [hunk 1]
```diff
@@ -248,9 +248,11 @@ var stylesRewriters = {
     //TODO: NATE: We probably want to map p tags to divs, it might be done already but I'm not 100% sure
 };
 
-var allowedBlock = /^(?:A(?:DDRESS|RTICLE|SIDE|UDIO)|BLOCKQUOTE|CAPTION|D(?:[DLT]|IV)|F(?:IGURE|IGCAPTION|OOTER)|H[1-6]|HEADER|L(?:ABEL|EGEND|I)|O(?:L|UTPUT)|P(?:RE)?|SECTION|T(?:ABLE|BODY|D|FOOT|H|HEAD|R)|UL)$/;
+var allowedBlock = /^(?BLOCKQUOTE|D(?:[DLT]|IV)|H[1-3]|LI|OL|PRE|T(?:ABLE|BODY|D|FOOT|H|HEAD|R)|UL)$/;
+
+var blacklist = /^(?:ABBR|ADDRESS|APPLET|AREA|ARTICLE|ASIDE|AUDIO|BASE|BASEFONT|BDI|BDO|BGSOUND|BUTTON|CANVAS|CODE|COL|COLGROUP|COMMAND|CONTENT|DATA|DATALIST|DEL|DFN|DIALOG|DIR|ELEMENT|EMBED|FIELDSET|FIGURE|FONT|FORM|FRAME|FRAMESET|HEAD|HEADER|HR|IFRAME|IMAGE|IMG|INPUT|INS|ISINDEX|KBD|KEYGEN|LABEL|LEGEND|LINK|LISTING|MAIN|MAP|MARK|MARQUEE|MATH|MENU|MENUITEM|META|METER|MULTICOL|NAV|NOBR|NOEMBED|NOFRAMES|NOSCRIPT|OBJECT|OPTGROUP|OPTION|OUTPUT|PARAM|PRE|PICTURE|PROGRESS|RP|RT|RTC|RUBY|SAMP|SCRIPT|SELECT|SHADOW|SOURCE|SPACER|STYLE|SVG|TEMPLATE|TEXTAREA|TFOOT|TIME|TITLE|TRACK|VAR|VIDEO|WBR|XMP)/;
```
Comment (dginev) [inline source/Clean.js]: could youa add a `$` at the end of this regex? A bit more precise, the old one should've had it too

**Comment by dginev** [comment]
This PR will make me sleep better at night, big :+1: from me as well. Our thinking for the moment is that it's better to err on the side of caution, and make sure we don't allow elements we don't currently use in Authorea's squire subset, while also blacklisting elements that look definitively foreign in the long-term.

**Comment by dginev** [comment]
General question to @natejenkins : do we need to version bump squire in every PR? In order to successfully deploy the newer version?

I think now that our package.json has an explicit `sha` of the squire repo, we can make due without the version bump (but we need to remember to update package.json in socialApp)

**Comment by dginev** [comment]
LGTM ASAP HURRY PLS

**Comment by DZGoldman** [comment]
https://sentry.authorea.com/sentry/client-production/issues/4596/events/161424/ Inline font tags are sneaking into our editor. The time is nigh

**Pull request merged**